core/src/main/java/io/questdb/cairo/wal/WalWriter.java

```diff
@@ -1333,20 +1366,35 @@ private void switchColumnsToNewSegment(LongList newColumnFiles) {
                 long currentOffset = newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 1);
                 long newOffset = newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 2);
                 primaryColumnFile.jumpTo(currentOffset);
+                primaryColumnFile.sync(false);
                 primaryColumnFile.switchTo(newPrimaryFd, newOffset, Vm.TRUNCATE_TO_POINTER);
+                primaryColumnFile.sync(false);
 
                 int newSecondaryFd = (int) newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 3);
                 if (newSecondaryFd > -1) {
                     MemoryMA secondaryColumnFile = getSecondaryColumn(i);
                     currentOffset = newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 4);
                     newOffset = newColumnFiles.get(i * NEW_COL_RECORD_SIZE + 5);
                     secondaryColumnFile.jumpTo(currentOffset);
+                    secondaryColumnFile.sync(false);
```

Comment:
It should be fine not to sync old column files, they should already be sync-ed in the main commit method.

Reply:
Refactored this place to not sync anything. We have to msync all mmap stuff and fsync file stuff. I moved the logic to where we copy columns. Tested for destruction, all works :)

Reply:
here: https://github.com/questdb/questdb/pull/3304/commits/7193f2eeba3eb1383f26852069b1678d4580d3b0